docs/projects/selection-sort/_posts/2019-01-10-python.md

```diff
@@ -0,0 +1,176 @@
+It also deals with any errors raised.
+
+### input_list()
+
+```python
+def input_list(list_str):
+    return [int(x.strip(" "), 10) for x in list_str.split(',')]
+```
+
+This function takes a string like `"2, 1, 10, 5, 3"`, and turns into a list of numbers. It does this using a list comprehension, first we need to convert
+our string into a list `list_str.split(',')` which is a list of strings split by comma (,). So our original input string becomes `["2", " 1", " 10", " 5", " 3"]`.
+Then for each element in the list `for x in ...` ,  we do something to it.
+
+In this example we convert it into a decimal integer, `int(x.strip(" "), 10)`
+`x.strip(" ")`, removes any whitespace so `" 1"` becomes `"1"`, then `int("1", 10)` converts the string `"1"` into a decimal number in this case very simply `1`. This is done
+for every item in the list so our original input of `"2, 1, 10, 5, 3"` becomes `[2, 1, 10, 5, 3]`.
+
+### exit_with_error()
+
+```python
+def exit_with_error():
+    print('Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”')
+    sys.exit(1)
+```
+
+This function simply prints a message and then exits the script with an error, `sys.exit(1)`. If any non-zero value is returned then the program didn't complete properly.
+This function is called if the user input isn't correct.
+
+### selection_sort()
+
+```python
+def selection_sort(xs, sorted_xs=None):
+    sorted_xs = sorted_xs or []
+    if len(xs) <= 0:
+        return sorted_xs
+    x = min(xs)
+    sorted_xs.append(x)
+    xs.remove(x)
+    return selection_sort(xs, sorted_xs)
+```
+
+Now onto the main part of the program, this is the function that actually sorts our list.
+The `selection_sort()` takes two parameters `xs` which is the unsorted list and `sorted_xs` which funnily enough is the current sorted list.
+When you first call the `selection_sort()` function you simply pass it your unsorted list as `sorted_xs=None` by default.
+
+If the `sorted_xs` value is set (not `None`) then we make `sorted_xs` equal itself, else `sorted_xs` equals `[]` (an empty list).
+You should never make a mutable object a default argument in Python as you get can get unexpected result. You can get more information [here](http://effbot.org/zone/default-values.htm). Therefore we set `sorted_xs=None` instead of `sorted_xs=[]`.
+
+Then we check if xs is empty (`<=0`), which would mean we have sorted every element, then we simply return the `sorted_xs` which is the sorted this.
+We can do this because as we sort element we move them from `xs` to `sorted_xs` ( items get removed from the `xs` list).
+
+If `xs` still has items then that means we haven't completely sorted the list. We found the smallest value in `x = min(xs)`,
+We append that value to `sorted_xs` and then we remove it from the `xs` list. Finally, we call the selection sort function with
```

Comment:
There's an issue with the commas in this sentence. Maybe the previous comma is supposed to be a period. I'm not sure.

Reply:
Hmm yeh I'll look at rewording the sentence